src/storage/txn/actions/flashback_to_version.rs

```diff
@@ -114,27 +123,32 @@ pub fn flashback_to_version_write(
         if txn.write_size() >= MAX_TXN_WRITE_SIZE {
             return Ok(Some(key.clone()));
         }
+        let old_write = reader.get_write(&key, flashback_version)?;
```

Comment:
So, now you scan out the keys that need flashback, and use `get_write` to get the old write and rewrite. Could it cause a lot more `seek`? I'm not sure about the overall performance...

Reply:
Actual, the old way is also `scan` and `seek`: https://github.com/tikv/tikv/blob/dc347f98ae70e8f829a55d4df1f002b4ad17e72d/src/storage/mvcc/reader/reader.rs#L624-L630

🤔 This PR should reduce the overhead since it will reduce the key number we need to scan and seek. And furthermore, maybe we should rewrite the `scan_writes` to make it not seek the old version if we don't need it to read the old value now.

Reply:
According to the current [implementation](https://github.com/tikv/tikv/blob/dc347f98ae70e8f829a55d4df1f002b4ad17e72d/src/storage/mvcc/reader/reader.rs#L625) of `scan_writes`, every key that needs to be flashbacked will also have a `seek` to find its latest visible write record before the given `version`, now since we don't need to get the record in `scan_writes` anymore, maybe we can rewrite it to only return the latest record without seeking. By this, the total number of `seek` should be kept as same as before.

Reply:
Or we can just make `scan_writes` become `scan_latest_keys`.

Reply:
> maybe we can rewrite it to only return the latest record without seeking

I doubt whether it's possible. In scan_write, after you get the latest version of one key, then how to move the cursor to the next key? near_seek is still the solution. In the current implementation, all the keys are scanned but each key is one near_seek. Now, all keys scanned need two near_seek but SSTs are filtered by hint.

Anyway, intuitionally this PR should win in real cases...

Reply:
I rewrote `scan_writes` to `scan_latest_user_keys`, and now it will only return all the latest user keys, since we no longer need the write record. PTAL.

BTW, a real-world performance test is in progress by @fengou1, the results will be updated later.